### PR TITLE
Closes #90 | Add Dataloader for UDHR-LID

### LINKED ISSUE #90: Create dataset loader for UDHR-LID
Dataloader name: `udhr_lid/udhr_lid.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?udhr_lid

|     Dataset| udhr_lid |
|-------------|---|
| Description | The UDHR-LID dataset is a refined version of the Universal Declaration of Human Rights, tailored for language identification tasks. It removes filler texts, repeated phrases, and inaccuracies from the original UDHR, focusing only on cleaned paragraphs. Each entry in the dataset is associated with a specific language, providing long, linguistically rich content. This dataset is particularly useful for non-parallel, language-specific text analysis in natural language processing. |
| Subsets | sun, ace, mad, lao, cfm, hnj, min, zlm, tha, blt, hni, jav, tdt, cnh, khm, ban, ind, mya, ccp, duu, tet, kkh, bug, vie |
| Languages | sun, ace, mad, lao, cfm, hnj, min, zlm, tha, blt, hni, jav, tdt, cnh, khm, ban, ind, mya, ccp, duu, tet, kkh, bug, vie |
| Tasks | Language Identification |
| License | Creative Commons Zero v1.0 Universal (cc0-1.0) |
| Homepage | https://huggingface.co/datasets/cis-lmu/udhr-lid |
| HF URL | https://huggingface.co/datasets/cis-lmu/udhr-lid |
| Paper URL | https://arxiv.org/abs/2310.16248 |

### PR DESCRIPTION
**Pull request by muhsatrio**
Closes #90 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/udhr_lid/udhr_lid.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/udhr_lid/udhr_lid.py`.

### CODE REVIEW

**Review by jamesjaya — COMMENTED** [review]
This looks ok to me + tested. I have some questions.

File: seacrowd/sea_datasets/udhr_lid/udhr_lid.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+_HOMEPAGE = "https://huggingface.co/datasets/cis-lmu/udhr-lid"
+
+_LICENSE = Licenses.CC0_1_0.value
+
+_URL = "https://huggingface.co/datasets/cis-lmu/udhr-lid/raw/main/udhr-lid.csv"
+
+_SUPPORTED_TASKS = [Tasks.TOKEN_LEVEL_LANGUAGE_IDENTIFICATION]
+
+_SOURCE_VERSION = "1.1.0"
```
[the remark this answers is not in the document]
Comment (muhsatrio) [reply seacrowd/sea_datasets/udhr_lid/udhr_lid.py:57]: I forgot to change it since I copied first from another dataloader file hehehehehe let me fix it

**Comment by muhsatrio** [comment]
Hi kak @jamesjaya, had been adjusted as your review yap. Kindly check again, thank you!

**Review by jamesjaya — APPROVED** [review]

File: seacrowd/sea_datasets/udhr_lid/udhr_lid.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+_CITATION = """\
+@inproceedings{
+  kargaran2023glotlid,
+  title={{GlotLID}: Language Identification for Low-Resource Languages},
+  author={Kargaran, Amir Hossein and Imani, Ayyoob and Yvon, Fran{\c{c}}ois and Sch{\"u}tze, Hinrich},
+  booktitle={The 2023 Conference on Empirical Methods in Natural Language Processing},
+  year={2023},
+  url={https://openreview.net/forum?id=dl4e3EBz5j}
+}
+"""
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/udhr_lid/udhr_lid.py:35]: Line 30 gives an invalid escape sequence, maybe make this a raw string?
```suggestion
_CITATION = r"""\
@inproceedings{
  kargaran2023glotlid,
  title={{GlotLID}: Language Identification for Low-Resource Languages},
  author={Kargaran, Amir Hossein and Imani, Ayyoob and Yvon, Fran{\c{c}}ois and Sch{\"u}tze, Hinrich},
  booktitle={The 2023 Conference on Empirical Methods in Natural Language Processing},
  year={2023},
  url={https://openreview.net/forum?id=dl4e3EBz5j}
}
"""
```

File: seacrowd/sea_datasets/udhr_lid/udhr_lid.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+_LANGUAGES = ["sun", "ace", "mad", "lao", "cfm", "hnj", "min", "zlm", "tha", "blt", "hni", "jav", "tdt", "cnh", "khm", "ban", "ind", "mya", "ccp", "duu", "tet", "kkh", "bug", "vie"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+_LOCAL = False
+
+_DATASETNAME = "udhr_lid"
+
+_DESCRIPTION = """\
+The UDHR-LID dataset is a refined version of the Universal Declaration of Human Rights, tailored for language identification tasks. 
+It removes filler texts, repeated phrases, and inaccuracies from the original UDHR, focusing only on cleaned paragraphs. 
+Each entry in the dataset is associated with a specific language, providing long, linguistically rich content. 
+This dataset is particularly useful for non-parallel, language-specific text analysis in natural language processing.
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/udhr_lid/udhr_lid.py:46]: Trailing whitespaces

```suggestion
The UDHR-LID dataset is a refined version of the Universal Declaration of Human Rights, tailored for language identification tasks.
It removes filler texts, repeated phrases, and inaccuracies from the original UDHR, focusing only on cleaned paragraphs.
Each entry in the dataset is associated with a specific language, providing long, linguistically rich content.
This dataset is particularly useful for non-parallel, language-specific text analysis in natural language processing.
```

**Review by ljvmiranda921 — COMMENTED** [review]
Hi @muhsatrio , thanks for this PR! Data loader works well when tested. 

Just a few nits: I suggest running this first to the formatter so that it's consistent with other implementations:

```
make check_file=seacrowd/sea_datasets/udhr_lid/udhr_lid.py
```

**Comment by ljvmiranda921** [comment]
Hi @muhsatrio , let me know if you need help making the changes!

**Comment by muhsatrio** [comment]
Hi kak @ljvmiranda921, sorry just have time again to do the task hehehehe had adjusted as you requested yap. Please check again, thank you!

**Review by ljvmiranda921 — APPROVED** [review]
LGTM!

**Pull request merged**